system/behavior.lua

```diff
@@ -65,25 +68,50 @@ function Behavior:Initialize(isReload)
   self:ReportLoadedBehaviors()
 end
 
+function Behavior:DecideBestSpecialization()
+  if wector.CurrentScript.Game == 'wow_retail' then
+    return Me.Talents.ActiveSpecializationId
+  elseif wector.CurrentScript.Game == 'wow_wrath' then
+    local bestspec = -1
+    local bestspecpoints = -1
+    for _, v in pairs(Me.Talents.ActiveTalentGroup.Tabs) do
+      if v.Points > bestspecpoints then
+        bestspec = v.Id
+        bestspecpoints = v.Points
+      end
+    end
+    return bestspec
+  end
+  return -1
+end
+
 function Behavior:Update()
-  local behavior = self.Active
-  -- if no behavior is active, return
-  if not behavior then return end
+  -- Use class_trim, as it seems to be the variable containing the player's class in lowercase
+  local class_trim = Me.ClassName:gsub("%s+", "")
+  class_trim = class_trim:lower()
+
+  local class = class_trim
+  local specid = self:DecideBestSpecialization()
+  local specname = behavior_map[class][specid]
 
-  -- Call all behaviors in whatever order they are in
-  -- Could sort to call them in a specific order
+  -- Loop through BehaviorTypes
   for _, k in pairs(BehaviorType) do
-    if behavior.Behaviors[k] then
-      behavior.Behaviors[k]()
+    -- Check if a callback exists for the class, specname, and behavior type
```

Comment:
after you revert the sanity check at the beginning of the function, do `behavior.Callbacks[class]` etc